src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -96,6 +97,17 @@ public override async Task StopAsync(CancellationToken cancellationToken)
             await base.StopAsync(cancellationToken);
         }
 
+        /// <summary>
+        /// Try to process a single message after the circuit was broken, a.k.a entering the half-open state.
+        /// </summary>
+        /// <returns>
+        ///     [true] when the related message handler can again process messages and the message pump can again start receive messages in full; [false] otherwise.
+        /// </returns>
+        public virtual Task<bool> TryProcessProcessSingleMessageAsync(MessagePumpCircuitBreakerOptions options)
```

Comment:
Maybe we should change this `bool` return type, so we can later incorporate the message count and have the message circuit breaker respond to this. As I imagine we will need more return types than success/failure.

Reply:
I think that's a good idea from an extensibility point of view.

Right now, a boolean value that indicates success or failure might be enough, but if we need to return more information, we'll need to change this to another type that contains that information.
So I guess changing the return type to a type like `MessageProcessingResult' that contains -for now- a boolean property that indicates success or failure is a good idea.

Reply:
Used a `MessageProcessingResult` in https://github.com/arcus-azure/arcus.messaging/pull/416/commits/e547cc0b377d3157280d7dff7f39ee71cf16a14e